Silicon/CommonSocPkg/Library/HeciLib/HeciLib.c

```diff
@@ -182,17 +184,36 @@ HeciGetFwCapsSkuMsg (
   EFI_STATUS                      Status;
   UINT32                          Length;
   GEN_GET_FW_CAPSKU              *MsgGenGetFwCapsSku;
-  GEN_GET_FW_CAPS_SKU_ACK        *MsgGenGetFwCapsSkuAck;
+  GEN_GET_FW_CAPS_SKU_ACK_DATA   *MsgGenGetFwCapsSkuAck;
+  ME_BIOS_PAYLOAD                *MbpDataHob;
+  UINT32                          MbpDataHobLen;
+  UINT8                          *DataPtr;
+  VOID                           *FspHobPtrList;
 
   if ((MsgGetFwCaps == NULL) || (MsgGetFwCapsAck == NULL) ) {
     return EFI_INVALID_PARAMETER;
   }
 
   MsgGenGetFwCapsSku    = (GEN_GET_FW_CAPSKU *)MsgGetFwCaps;
-  MsgGenGetFwCapsSkuAck = (GEN_GET_FW_CAPS_SKU_ACK *)MsgGetFwCapsAck;
+  MsgGenGetFwCapsSkuAck = (GEN_GET_FW_CAPS_SKU_ACK_DATA *)MsgGetFwCapsAck;
 
   //
-  // Allocate MsgGenGetFwVersion data structure
+  // Fill the FW capapbility data.
+  //
+  FspHobPtrList = GetFspHobListPtr();
+  if (FspHobPtrList != NULL) {
+    DataPtr = (UINT8 *)GetGuidHobData (GetFspHobListPtr(), &MbpDataHobLen, &gMeBiosPayloadHobGuid);

```

Comment:
why do we call GetFspHobListPtr() again ?   We should use FspHobList variable, right ?

Reply:
yeah..my bad. change done.